services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -436,29 +436,12 @@ public RoutesToNeutronWebResponse removeRoutesFromNeutronRouter(String routerid,
         }
         RouteTable routeTable = router.getNeutronRouteTable();
         List<RouteEntry> routeEntities = routeTable.getRouteEntities();
-
         List<NewRoutesRequest> requestRoutes = requestRouter.getRoutes();
-        // TODO: time complexity O(n^2), check if it effect performance
-        for (NewRoutesRequest requestRoute : requestRoutes) {
-            String requestDestination = requestRoute.getDestination();
-            String requestNexthop = requestRoute.getNexthop();
 
-            if (requestDestination == null || requestNexthop == null) {
-                throw new DestinationOrNexthopCanNotBeNull();
-            }
-
-            for (int i = 0; i < routeEntities.size(); i++) {
-                RouteEntry routeEntry = routeEntities.get(i);
-                String destination = routeEntry.getDestination();
-                String nexthop = routeEntry.getNexthop();
-                if (destination.equals(requestDestination) && nexthop.equals(requestNexthop)) {
-                    routeEntities.remove(i);
-                    break;
-                }
-            }
-        }
-        routeTable.setRouteEntities(routeEntities);
-        router.setNeutronRouteTable(routeTable);
+        // TODO: time complexity O(n^2), check if it effect performance
+        requestRoutes.forEach(item -> {
+            routeEntities.removeIf(routingRule -> routingRule.getDestination().equals(item.getDestination()) && routingRule.getNexthop().equals(item.getNexthop()));
+        });
```

Comment:
@DavidLiu506 Good compact code to simply two-for-loops. Do you need write `routeEntities` back to the `router` structure after the process done here?

Reply:
No, it is a reference.